saw-core/src/Verifier/SAW/Simulator.hs

```diff
@@ -142,10 +142,10 @@ evalTermF cfg lam recEval tf env =
                                     if inBitSet 0 (looseVars t2) then
                                       pure (VDependentPi (\x -> toTValue <$> lam t2 ((x,v) : env)))
                                     else
-                                      do val <- delay (panic "evalTermF"
-                                                         ["nondependent Pi type forced its value"
-                                                         , showTerm (Unshared tf)])
-                                         VNondependentPi . toTValue <$> lam t2 ((val,v) : env)
+                                      do -- put dummy values in the environment; the term should never reference them
+                                         let val = ready VUnit
+                                         let tp  = VUnitType
```

Comment:
Putting a term with the wrong type in the environment seems a bit icky to me. (Granted, hiding a `panic` in there was already icky.) Can you think of any cleaner fixes for this? Perhaps the environment could be a list of `Maybe`s, and we could just insert a `Nothing`?

Reply:
I guess we could do that, but it adds an additional indirection and pattern match on the common case.

I claim we can put any type into the context that we like, provided the variable doesn't appear in the term; and that this follows as a pretty standard consequence of weakening.

Reply:
Yeah, I guess the need to do a pattern match is not nice. I suppose it's fine how it is.